directory_sso_api_client/models.py

```diff
@@ -0,0 +1,28 @@
+from django.contrib.auth.models import AbstractUser
+
+from django.db import models
+
+from directory_sso_api_client import sso_api_client
+
+
+class SSOUser(AbstractUser):
+    session_id = models.TextField()
+    hashed_uuid = models.CharField(max_length=200)
+
+    def check_password(self, raw_password):
+        response = sso_api_client.usercheck_password(
+            session_id=self.session_id, password=raw_password
+        )
+        return response.ok
+
+    def get_username(self):
+        return self.email
+
+    def set_password(self):
+        raise NotImplementedError
+
+    def set_unusable_password(self):
+        raise NotImplementedError
+
+    def save(self, *args, **kwargs):
+        pass
```

Comment:
Would `raise NotImplementedError` be better here?
I think I'd prefer it to obviously break than silently do nothing - but perhaps I'm missing a reason that this is a bad idea!

Reply:
django wants to save `last_logged_in` - resulting in `NotImplementedError`. I've now added a comment to save to indicate why we're using `pass`.